### DISCUSSION TITLE
textureCompress on client side

### DISCUSSION THREAD

**Discussion by Search-For-a-Friend** [donmccurdy/glTF-Transform#973] [Q&A] [answered]
Hello and thank you for this amazing tool !

I am working on a Web App for sharing 3D assets. I looked for solution to manipulate and simplify 3D models from Client side, before uploading them to the server.
I found your library could handle some features from Client side, however, it seems textureCompress is requiring sharp which is Node.js environment only.

Do you suggest a workaround for using your tool with another textureCompression process ?

Great thanks !

**Comment by donmccurdy** [comment, marked as answer]
Hi @Search-For-a-Friend! Unfortunately I don't have a workaround for client-side texture compression right now. There appears to be some progress toward a WebAssembly version of Sharp (https://github.com/lovell/sharp/pull/3522) which would not immediately mean browser support, but might eventually lead there. You might also look at https://github.com/kleisauke/wasm-vips or other libraries. If you find a library that supports the environment and compression methods that you need, using that with glTF Transform shouldn't be too difficult. See the discussion below for how you might write a function applying your own compression in glTF Transform:

- https://github.com/donmccurdy/glTF-Transform/issues/845#issuecomment-1467364895

**Comment by Search-For-a-Friend** [comment]
Thank you for this detailed workaround, as I don't have strong requirements for the compression itself I am sure I will come up with something satisfying with the leads you gave me.
Great thanks for your time, I might let you know if something still troubles me when using your library on web client.
